src/iroh.rs

```diff
@@ -474,159 +422,281 @@ impl VeilidIrohBlobs {
         // Upload the file stream and get its hash
         let file_hash = self.upload_from_stream(file_stream).await?;
 
         // Add the uploaded file to the collection
-        self.set_file(collection_name, path, file_hash).await
+        self.set_file(&collection_name, &path, &file_hash).await
     }
     pub async fn collection_hash(&self, collection_name: &str) -> Result<Hash> {
-        let map = self.collection_hashes.read().await;
-        if let Some(hash) = map.get(collection_name) {
-            Ok(*hash)
-        } else {
-            Err(anyhow!("Collection {} not found", collection_name))
+        // Retrieve the tag from the store instead of using the in-memory cache
+        match self.get_tag(collection_name).await {
+            Ok(hash) => Ok(hash),
+            Err(e) => {
+                println!("Error retrieving tag for collection: {:?}", e);
+                Err(anyhow!("Collection {} not found", collection_name))
+            }
+        }
+    }
+    pub async fn store_tag(&self, collection_name: &str, collection_hash: &Hash) -> Result<()> {
+        println!("Storing tag for collection: {} with hash: {:?}", collection_name, collection_hash);
+        
+        // Store the tag
+        self.store.set_tag(
+            collection_name.to_string().into(), 
+            Some(HashAndFormat::new(*collection_hash, BlobFormat::Raw))
+        ).await?;
+    
+        // Introducing a short delay to ensure that tag is persisted
+        tokio::time::sleep(Duration::from_millis(100)).await;
+    
+        // Verify that the tag was actually stored
+        match self.get_tag(collection_name).await {
+            Ok(stored_hash) => {
+                if stored_hash == *collection_hash {
+                    println!("Tag successfully stored for collection: {}", collection_name);
+                } else {
+                    return Err(anyhow!("Mismatch in stored tag for collection: {}", collection_name));
+                }
+            },
+            Err(e) => {
+                return Err(anyhow!("Failed to retrieve tag after storing for collection: {}. Error: {:?}", collection_name, e));
+            }
+        }
+        
+        Ok(())
+    }
+    
+    
+    
+    
+    pub async fn get_tag(&self, collection_name: &str) -> Result<Hash> {
+        println!("Retrieving tag for collection: {}", collection_name);
+        let mut tags = self.store.tags().await?;
+    
+           
+        while let Some(tag_result) = tags.next() {
+            match tag_result {
+                Ok((tag, hash_and_format)) => {
+                    // Logging for better insight
+                    println!("Checking tag: {:?}", tag);
+
+                    // Extract the bytes and convert them to a string
+                    let tag_bytes = tag.borrow();
+                    let tag_str = std::str::from_utf8(tag_bytes).unwrap_or("");
```

Comment:
Instead lets conver the name to byes and do a `tag_bytes.eq` to save on parsing

Reply:
I updated get_tag to compare bytes to save on parsing
dc0b4b490faaf22a16bf5d539afa5393be33e901